src/cache.ts

```diff
@@ -35,25 +35,25 @@ const restoreCache = (cachePath: string, cacheKey: string) => {
   })
 }
 
-const generateEnvironmentKey = (options: Options, prefix: string) => {
+const generateEnvironmentKey = async (options: Options, prefix: string) => {
   const arch = `-${getCondaArch()}`
   const envName = options.environmentName ? `-${options.environmentName}` : ''
   const createArgs = options.createArgs ? `-args-${sha256Short(JSON.stringify(options.createArgs))}` : ''
   const rootPrefix = `-root-${sha256Short(options.micromambaRootPath)}`
-  const binHash = fs.readFile(options.micromambaBinPath).then(sha256)
+  const binHash = await fs.readFile(options.micromambaBinPath, 'utf-8').then(sha256)
```

Comment:
Reading this binary executable as UTF-8 seems wrong.

Reply:
Great point, wasn't thinking. Should just hash as bytes